diff --git a/libraries/exoplayer/src/main/java/androidx/media3/exoplayer/AudioFocusManager.java b/libraries/exoplayer/src/main/java/androidx/media3/exoplayer/AudioFocusManager.java
index f865c72e1d..91a05ba139 100644
--- a/libraries/exoplayer/src/main/java/androidx/media3/exoplayer/AudioFocusManager.java
+++ b/libraries/exoplayer/src/main/java/androidx/media3/exoplayer/AudioFocusManager.java
@@ -130,6 +130,14 @@ import org.checkerframework.checker.nullness.qual.MonotonicNonNull;
 
   private @AudioFocusState int audioFocusState;
   private @AudioManagerCompat.AudioFocusGain int focusGainToRequest;
+
+  public boolean inu_transientFocus;
+
+  private @AudioManagerCompat.AudioFocusGain int inu_getFocusGainToRequest() {
+    return inu_transientFocus && focusGainToRequest == AUDIOFOCUS_GAIN
+        ? AUDIOFOCUS_GAIN_TRANSIENT
+        : focusGainToRequest;
+  }
   private float volumeMultiplier = VOLUME_MULTIPLIER_DEFAULT;
   private @MonotonicNonNull AudioFocusRequestCompat audioFocusRequest;
   private boolean rebuildAudioFocusRequest;
@@ -248,7 +256,7 @@ import org.checkerframework.checker.nullness.qual.MonotonicNonNull;
     if (audioFocusRequest == null || rebuildAudioFocusRequest) {
       AudioFocusRequestCompat.Builder builder =
           audioFocusRequest == null
-              ? new AudioFocusRequestCompat.Builder(focusGainToRequest)
+              ? new AudioFocusRequestCompat.Builder(inu_getFocusGainToRequest())
               : audioFocusRequest.buildUpon();
 
       boolean willPauseWhenDucked = willPauseWhenDucked();
diff --git a/libraries/exoplayer/src/main/java/androidx/media3/exoplayer/ExoPlayerImpl.java b/libraries/exoplayer/src/main/java/androidx/media3/exoplayer/ExoPlayerImpl.java
index 50de205709..55ce8fa551 100644
--- a/libraries/exoplayer/src/main/java/androidx/media3/exoplayer/ExoPlayerImpl.java
+++ b/libraries/exoplayer/src/main/java/androidx/media3/exoplayer/ExoPlayerImpl.java
@@ -128,7 +128,7 @@ import java.util.concurrent.CopyOnWriteArraySet;
 import java.util.concurrent.Executor;
 
 /** The default implementation of {@link ExoPlayer}. */
-/* package */ final class ExoPlayerImpl extends BasePlayer implements ExoPlayer {
+public final class ExoPlayerImpl extends BasePlayer implements ExoPlayer {
 
   static {
     MediaLibraryInfo.registerModule("media3.exoplayer");
@@ -1508,6 +1508,14 @@ import java.util.concurrent.Executor;
     listeners.flushEvents();
   }
 
+  public void inu_setTransientAudioFocus(boolean transientFocus) {
+    verifyApplicationThread();
+    if (playerReleased) {
+      return;
+    }
+    internalPlayer.inu_setTransientAudioFocus(transientFocus);
+  }
+
   @Override
   public AudioAttributes getAudioAttributes() {
     verifyApplicationThread();
diff --git a/libraries/exoplayer/src/main/java/androidx/media3/exoplayer/ExoPlayerImplInternal.java b/libraries/exoplayer/src/main/java/androidx/media3/exoplayer/ExoPlayerImplInternal.java
index baefc5c3be..abe928ab7e 100644
--- a/libraries/exoplayer/src/main/java/androidx/media3/exoplayer/ExoPlayerImplInternal.java
+++ b/libraries/exoplayer/src/main/java/androidx/media3/exoplayer/ExoPlayerImplInternal.java
@@ -654,6 +654,10 @@ import java.util.Objects;
     handler.obtainMessage(MSG_AUDIO_FOCUS_PLAYER_COMMAND, playerCommand, 0).sendToTarget();
   }
 
+  public void inu_setTransientAudioFocus(boolean transientFocus) {
+    audioFocusManager.inu_transientFocus = transientFocus;
+  }
+
   // VideoFrameMetadataListener implementation
 
   @Override
