From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: The Ghost <lurepheonix@gmail.com>
Date: Thu, 27 Aug 2026 11:07:36 +0400
Subject: Make update channel configurable via build config

---
 TMessagesProj/build.gradle | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/TMessagesProj/build.gradle b/TMessagesProj/build.gradle
index 0000000..0000000 100644
--- a/TMessagesProj/build.gradle
+++ b/TMessagesProj/build.gradle
@@ -162,6 +162,8 @@ android {
             buildConfigField "boolean", "DEBUG_VERSION", "true"
             buildConfigField "boolean", "DEBUG_PRIVATE_VERSION", "true"
             buildConfigField "boolean", "BUNDLE", "false"
+            buildConfigField "String", "UPDATE_CHANNEL_USERNAME", "\"" + (System.getenv('UPDATE_CHANNEL_USERNAME') ?: 'InugramCI') + "\""
+            buildConfigField "long", "UPDATE_CHANNEL_ID", (System.getenv('UPDATE_CHANNEL_ID') ?: '3968318575L')
         }
 
         release {
@@ -178,6 +180,8 @@ android {
             buildConfigField "boolean", "DEBUG_VERSION", "false"
             buildConfigField "boolean", "DEBUG_PRIVATE_VERSION", "false"
             buildConfigField "boolean", "BUNDLE", "false"
+            buildConfigField "String", "UPDATE_CHANNEL_USERNAME", "\"" + (System.getenv('UPDATE_CHANNEL_USERNAME') ?: 'InugramCI') + "\""
+            buildConfigField "long", "UPDATE_CHANNEL_ID", (System.getenv('UPDATE_CHANNEL_ID') ?: '3968318575L')
         }
     }
 
